community/server/src/main/java/org/neo4j/server/configuration/ConfigLoader.java

```diff
@@ -96,8 +96,10 @@ private static void overrideEmbeddedDefaults( HashMap<String, String> config )
         config.putIfAbsent( ShellSettings.remote_shell_enabled.name(), TRUE );
         config.putIfAbsent( GraphDatabaseSettings.logs_directory.name(), "logs" );
         config.putIfAbsent( GraphDatabaseSettings.auth_enabled.name(), "true" );
+
+        String dataDirectory = config.getOrDefault( data_directory.name(), data_directory.getDefaultValue() );
         config.putIfAbsent( GraphDatabaseSettings.auth_store.name(),
-                new File( config.get( data_directory.name() ), "dbms/auth" ).toString() );
+                new File( dataDirectory, "dbms/auth" ).toString() );
     }
```

Comment:
What about if we set `data_directory` default, as we've set `logs` above? I'd prefer not frigging with an internal setting.